config/seata-config-consul/src/main/java/io/seata/config/consul/ConsulConfiguration.java

```diff
@@ -0,0 +1,230 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.config.consul;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.QueryParams;
+import com.ecwid.consul.v1.Response;
+import com.ecwid.consul.v1.kv.model.GetValue;
+import com.ecwid.consul.v1.kv.model.PutParams;
+import io.seata.common.thread.NamedThreadFactory;
+import io.seata.config.AbstractConfiguration;
+import io.seata.config.ConfigChangeListener;
+import io.seata.config.ConfigFuture;
+import io.seata.config.Configuration;
+import io.seata.config.ConfigurationFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static io.seata.config.ConfigurationKeys.FILE_CONFIG_SPLIT_CHAR;
+import static io.seata.config.ConfigurationKeys.FILE_ROOT_REGISTRY;
+
+/**
+ * @author xingfudeshi@gmail.com
+ * @date 2019/05/05
+ */
+public class ConsulConfiguration extends AbstractConfiguration<ConfigChangeListener> {
+    private volatile static ConsulConfiguration instance;
+    private volatile static ConsulClient client;
+
+    private static final Configuration FILE_CONFIG = ConfigurationFactory.FILE_INSTANCE;
+    private static final String SERVER_ADDR_KEY = "serverAddr";
+    private static final String REGISTRY_TYPE = "consul";
+    private static final String FILE_CONFIG_KEY_PREFIX = FILE_ROOT_REGISTRY + FILE_CONFIG_SPLIT_CHAR + REGISTRY_TYPE + FILE_CONFIG_SPLIT_CHAR;
```

Comment:
why read registry node？

Reply:
Registry and config are completely independent

Reply:
> Registry and config are completely independent

By mistake,will correct it.

Reply:
> Registry and config are completely independent

thanks.it fixed.